### ISSUE TITLE
`contrasted_opening_brace` false positives in trailing closures and `else if` statements

### ISSUE THREAD

**Issue by buresdv** [realm/SwiftLint#5752]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

I'm using the new `contrasted_opening_brace` rule. I ran across a potential false positive (at least in my opinion) when this rule is used on trailing closures that include parameters.

Examples include (lifted straight from my codebase):
```swift
return allDependencies.map
{ dependency in // <-- Violation here
    .init(name: dependency.fullName, version: dependency.version, directlyDeclared: dependency.declaredDirectly)
}
```

```swift
let originalServicesWithTheirUUIDs: [String: UUID] = services.reduce(into: [:])
{ result, originalService in // <-- Violation here
    result[originalService.name] = originalService.id
}
```

```swift
return AsyncStream
{ continuation in // <-- Violation here
}
```

I'd expect this rule to not consider these examples violations. Perhaps there could be a setting that would not flag these.

**Comment by buresdv** [comment]
Another potentionally related false positive occurs in long else-if chains (don't judge me 😅 )
![Snímek obrazovky 2024-08-14 v 22 44 06](https://github.com/user-attachments/assets/09c6a5f4-033c-4aba-b4ef-9e755e1e9978)

**Comment by buresdv** [comment]
@SimplyDanny you're the man

**Comment by SimplyDanny** [comment]
@buresdv: Formatting rules often rely on heuristics and so does this rule now especially in terms of the fixes I made to cover your reported misbehavior. So you might find other issues afterwards which we need to address in an iterative manner.
